source/common/http/match_delegate/config.cc

```diff
@@ -286,6 +303,34 @@ Envoy::Http::FilterFactoryCb MatchDelegateConfig::createFilterFactoryFromProtoTy
   };
 }
 
+Router::RouteSpecificFilterConfigConstSharedPtr
+MatchDelegateConfig::createRouteSpecificFilterConfigTyped(
+    const envoy::extensions::common::matching::v3::ExtensionWithMatcherPerRoute& proto_config,
+    Server::Configuration::ServerFactoryContext& server_context,
+    ProtobufMessage::ValidationVisitor&) {
+  return std::make_shared<FilterConfigPerRoute>(proto_config, server_context);
+}
+
+Matcher::MatchTreeSharedPtr<Envoy::Http::HttpMatchingData>
+FilterConfigPerRoute::createFilterMatchTree(
+    const envoy::extensions::common::matching::v3::ExtensionWithMatcherPerRoute& proto_config,
+    Server::Configuration::ServerFactoryContext& server_context) {
+  auto requirements =
+      std::make_unique<envoy::extensions::filters::common::dependency::v3::MatchingRequirements>();
+  Server::Configuration::FactoryContext* context =
+      reinterpret_cast<Server::Configuration::FactoryContext*>(&server_context);
```

Comment:
This is a temporary workaround, not intended to be merged into main.

Reply:
#27621 implements the formal solution for this. (It is ready for review I just need to add the PR description)

Reply:
yeah, I think we cann't merge it before this to be addressed first.